pkg/injector/envoy_container.go

```diff
@@ -15,21 +15,36 @@ const (
 	envoyProxyConfigPath     = "/etc/envoy"
 )
 
-func getEnvoySidecarContainerSpec(pod *corev1.Pod, cfg configurator.Configurator, originalHealthProbes healthProbes) corev1.Container {
+func getPlatformSpecificSpecComponents(cfg configurator.Configurator, podOS string) (*corev1.SecurityContext, string) {
```

Comment:
nit: with multiple return values (where value type != error), it's always good to add a function comment regarding the return values, or use named return values such as https://tour.golang.org/basics/7.